aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+            option::some(maximum_supply) // limited supply
+        } else {
+            option::none() // unlimited supply
+        };
+
+        // creates named object (address of object is created using `creator address` and user generated `seed`) and returns the ConstructorRef
+        // `create_primary_store_enabled_fungible_asset` creates a fungible asset with primary store support by doing the following:
+        // ConstructorRef is used to generate an object signer, to create `Metadata` resource and store at the `Metadata object's` address, 
+        // and also store `DeriveRefPod` Resource within the object (at the same address)
+        let constructor_ref = &object::create_named_object(owner, ASSET_SYMBOL);
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            constructor_ref,
+            supply,
+            name,
+            symbol,
+            decimals,
+            string::utf8(b""),
+            string::utf8(b"")
+        );
+
+        // generate MintRef, BurnRef, and TransferRef using ConstructorRef
+        let mint_ref = fungible_asset::generate_mint_ref(constructor_ref);
+        let burn_ref = fungible_asset::generate_burn_ref(constructor_ref);
+        let transfer_ref = fungible_asset::generate_transfer_ref(constructor_ref);
+
+        // store refs in `FiatTokenConfig` resource, move resource to `Metadata object` address
+        let metadata_object_signer = object::generate_signer(constructor_ref);
+        move_to(
+            &metadata_object_signer,
+            FiatTokenConfig { mint_ref, burn_ref, transfer_ref },
+        );
+        // store access info in `FiatTokenAccessInfo` resource, move resource to `Circle/Owner` address
+        move_to(
+            owner,
+            FiatTokenAccessInfo {
+                minters: smart_table::new(),
+                blacklist: vector::empty(),
+            },
+        );
+        // store master minter in `MasterMinter` resource, move resource to `Circle/Owner` address
+        move_to(
+            owner,
+            MasterMinterConfig { master_minter },
+        )
+    }
+    
+    /// Mint new fiat tokens and deposit them to the `to` address.
+    /// Returns true if operation is successful.
+    public entry fun mint(minter: &signer, to: address, amount: u64) acquires FiatTokenConfig, FiatTokenAccessInfo  {
+        // SECURITY TODO: check if contract is paused 
+        // SECURITY TODO: check if transaction authorizer (signer) is a minter OR owner
+
+        // check if `minter` or `to` address is blacklisted
+        assert!(!is_blacklisted(signer::address_of(minter)), error::permission_denied(EBLACKLISTED_ADDRESS));
+        assert!(!is_blacklisted(to), error::permission_denied(EBLACKLISTED_ADDRESS));
+        
+        // check if amount is greater than the minter's allowance 
+        assert!(get_minter_allowance(signer::address_of(minter)) > amount, error::permission_denied(EMINTER_ALLOWANCE_EXCEEDED));
```

Comment:
You probably want `>=` here, since if the allowance is exactly the same, this will error right now.

May also want to use `invalid_input` instead of `permission_denied` for exceeding the allowance.